Pyfakefs 5.7.2 regression for NamedTemporaryFile

**Describe the bug**
It seems that pause/resume functionallity somehow causes NamedTemporaryFile to be double deleted, leading to test failures. This happens in `pyfakefs==5.7.2` but not `pyfakefs==5.7.1`.

**How To Reproduce**
Example test:

```py
from tempfile import NamedTemporaryFile


def test(fs):
    fs.pause()
    fs.resume()
    with NamedTemporaryFile() as f:
        pass
```

With crash that looks like this:

```
$ pytest test.py
============================= test session starts ==============================
platform darwin -- Python 3.11.10, pytest-8.3.3, pluggy-1.5.0
rootdir: 
configfile: pyproject.toml
plugins: pyfakefs-5.7.2, ...
collected 1 item

test.py F                                                                [100%]

=================================== FAILURES ===================================
_____________________________________ test _____________________________________

fs = <pyfakefs.fake_filesystem.FakeFilesystem object at 0x1129c02d0>

    def test(fs):
        fs.pause()
        fs.resume()
>       with NamedTemporaryFile() as f:

.../test.py:7: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.../3.11/lib/python3.11/tempfile.py:519: in __exit__
    self.close()
.../3.11/lib/python3.11/tempfile.py:526: in close
    self._closer.close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tempfile._TemporaryFileCloser object at 0x112982290>
unlink = <built-in function unlink>

    def close(self, unlink=_os.unlink):
        if not self.close_called and self.file is not None:
            self.close_called = True
            try:
                self.file.close()
            finally:
                if self.delete:
>                   unlink(self.name)
E                   FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/41/ptjvny1d7tsbn59t_ywv2y2m0000gq/T/tmpzytjbc5o'

.../3.11/lib/python3.11/tempfile.py:463: FileNotFoundError
=========================== short test summary info ============================
FAILED test.py::test - FileNotFoundError: [Errno 2] No such file or directory...
============================== 1 failed in 0.53s ===============================

```

**Your environment**

```bash
macOS-14.7.1-arm64-arm-64bit
Python 3.11.10 (main, Sep  7 2024, 01:03:31) [Clang 15.0.0 (clang-1500.3.9.4)]
pyfakefs 5.7.2
pytest 8.3.3
```

Note, I can also reproduce this on a linux VM.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -12,6 +12,12 @@ The released versions correspond to PyPI releases.
 * the default for `FakeFilesystem.shuffle_listdir_results` will change to `True` to reflect
   the real filesystem behavior
 
+## Unreleased
+
+### Fixes
+* fixed a regression in version 5.7.2 that `tempfile` was not patched after pause/resume
+  (POSIX only, see [#1098](../../issues/1098))
+
 ## [Version 5.7.2](https://pypi.python.org/pypi/pyfakefs/5.7.2) (2024-12-01)
 Fixes some problems with patching.
 

--- a/pyfakefs/fake_filesystem_unittest.py
+++ b/pyfakefs/fake_filesystem_unittest.py
@@ -125,6 +125,7 @@ class TempfilePatcher:
             tempfile._TemporaryFileCloser.close = self.tempfile_cleanup  # type: ignore[module-attr]
         else:
             tempfile._TemporaryFileCloser.cleanup = self.tempfile_cleanup  # type: ignore[module-attr]
+        self.tempfile_cleanup = None
         # reset the cached tempdir in tempfile
         tempfile.tempdir = None
 

--- a/pyfakefs/tests/fake_filesystem_unittest_test.py
+++ b/pyfakefs/tests/fake_filesystem_unittest_test.py
@@ -639,6 +639,13 @@ class PauseResumeTest(fake_filesystem_unittest.TestCase):
         with self.assertRaises(RuntimeError):
             fs.resume()
 
+    def test_that_tempfile_is_patched_after_resume(fs):
+        """Regression test for #1098"""
+        fs.pause()
+        fs.resume()
+        with tempfile.NamedTemporaryFile():
+            pass
+
 
 class PauseResumePatcherTest(fake_filesystem_unittest.TestCase):
     def test_pause_resume(self):

